framework/doc/content/source/problems/EigenProblem.md

```diff
@@ -2,16 +2,109 @@
+(e.g., an eigenvalue problem with Dirichlet boundary conditions).
+Many solvers in SLEPc can handle this linear eigenproblem.
+We can choose these algorithms in the executioner block.
+Here is an example with Jacobi Davidson:
+
+!listing test/tests/problems/eigen_problem/eigensolvers/gipm.i
+         block=Executioner
+         link=False
+
+Note that to use the linear solvers from SLEPc, we need to form accurate
+$A$ and $B$ that will be utilized to represent the underlying physics.
+If $A% and $B$ are incorrect, you will likely end with the
+wrong eigenvalue and eigenvector. For that reason, if you
+want to use linear eigenvalue solvers, you need to implement
+Jacobian by hand or with AD in the corresponding kernels.
+Details of all the linear eigensolvers can be found in
+[SLEPc Users Manual](https://slepc.upv.es/documentation/slepc.pdf).
+
+In a multiphysics environment, physics often is nonlinear. For instance,
+ neutron transport with temperature feedback. In this case,
+  $A$ and $B$ are not constant anymore; instead, they depend on
+   the eigenvector directly or indirectly. To describe our nonlinear
+    solver, we write a generalized nonlinear eigenvalue problem as,
+\begin{equation}
+A(x) = \lambda B(x),
+\end{equation}
+where $A(\cdot)$ and $B$ are nonlinear functions. In the EigenProblem
+system, the kernels of $A(\cdot)$ are referred to as "noneigen,"
+and the kernels of $B(\cdot)$ are referred to as "eigen."
+To utilize Newton to solve this problem, we define $\lambda = \frac{1}{|B(x)|}$,
+and rewrite the nonlinear eigen problem as
+\begin{equation}
+0=F(x) = A(x) - \lambda B(x).
+\end{equation}
+Newton can be applied to solve this problem. The only issue is that
+the derivatives of $\lambda$ with respective to $x$ can be dense.
+We use the Jacobian-free Newton method to overcome this difficulty,
+where the Jacobian is computed via finite difference.
+To handle various use cases, variants of Newton are provided
+in this system. There is a current list:
+
+- `PJFNK` - Preconditioned Jacobian-free Newton Krylov: The preconditioning matrix
+  is formed using nonegein kernels. We do have an option to allow users to
+  incorporate eigen kernels into the preconditioning matrix. Still, in
+  general, we won't encourage users to do that since the preconditioning
+  matrix might be singular when close to the solution.
+
+!listing test/tests/problems/eigen_problem/eigensolvers/ne.i
+          block=Executioner
+          link=False
+
+- `JFNK` - Jacobian-free Newton Krylov:  Users are responsible for
+  providing a preconditioner. This allows users to build a customized
+  preconditioner, such as a sweeper,
+  that might be more efficient for their underlying problems.
+
+!listing test/tests/problems/eigen_problem/preconditioners/ne_pbp.i
+          block=Executioner Preconditioning
+          link=False
+
+- `PJFNKMO`- Matrix-only Preconditioned Jacobian-free Newton Krylov: Mathematically,
```

Comment:
so this is 'linear-PJFNK' essentially?

Reply:
like the eigen-problem is linear?

Reply:
Yes, correctly.

Reply:
But it can do more than that.  If you have a nonlinear problem, this option allow you run more efficient residual evaluation via matrix-vector operations in  the linear step. But it could impact the convergence. 

There is a trade-off between iteration and compute time. It will be problem dependent